datastream-cloud-storage/src/main/java/com/linkedin/datastream/cloud/storage/io/ScribeParquetAvroConverter.java

```diff
@@ -0,0 +1,343 @@
+      return null;
+    }
+  }
+
+  /**
+   * The incoming records need to be converted to match the Parquet schema generated using
+   * `generateParquetStructuredAvroSchema` function. This function iterates all the fields in the avroRecord {@link ScribeParquetAvroConverter#generateParquetStructuredAvroSchema(Schema)}
+   * and converts them to necessary data types to conform to the target Parquet schema.
+   *
+   * @param schema parquet compatible avro schema
+   * @param avroRecord the incoming record
+   * @return GenericRecord the record converted to match the new schema
+   * @throws Exception
+   */
+  public static GenericRecord generateParquetStructuredAvroData(Schema schema, GenericRecord avroRecord) {
+    GenericRecord record = new GenericData.Record(schema);
+    Map<String,String> avroScribeHeaderFieldNamingMap = null;
+    GenericRecord scribeHeaderRecord = null;
+    if (avroRecord.get(SCRIBE_HEADER) != null) {
+      avroScribeHeaderFieldNamingMap = avroRecord.getSchema().getField(SCRIBE_HEADER).schema().getFields().stream()
+          .collect(Collectors.toMap(avroScribeHeaderField -> avroScribeHeaderField.name().toLowerCase(), avroScribeHeaderField -> avroScribeHeaderField.name()));
+      scribeHeaderRecord = getScribeHeaderParquetData(avroRecord, schema);
+    }
+
+    // Get fields from schema
+    List<Schema.Field> fields = schema.getFields();
+    List<Schema> avroFieldSchemaTypes = null;
+
+    // Map lowercase fieldnames to camelcase fieldnames.
+    // The fieldnames(ex: eventname) are in lowercase in parquet schema where as in avrorecord they are camelcase(eventName)
+    Map<String,String> avroFieldNamingMap = avroRecord.getSchema().getFields().stream()
+        .collect(Collectors.toMap(avroField-> avroField.name().toLowerCase(), avroField-> avroField.name()));
+
+    for (Schema.Field field : fields) {
+      String fieldName = avroFieldNamingMap.containsKey(field.name()) ? avroFieldNamingMap.get(field.name()) : field.name();
+      try {
+        if (fieldName!=null) {
+          if (avroRecord.get(fieldName) != null) {
+            avroFieldSchemaTypes = getSchemaTypes(fieldName, avroRecord);
+          } else if (avroRecord.get(fieldName) == null && ((GenericData.Record) avroRecord).getSchema().getField(fieldName) != null) {
+            continue;
+          } else if (avroScribeHeaderFieldNamingMap.containsKey(field.name()) && avroRecord.get(SCRIBE_HEADER) != null
+              && avroRecord.get(SCRIBE_HEADER).getClass().getSimpleName().equalsIgnoreCase("Record")) {
+            avroFieldSchemaTypes = getSchemaTypes(SCRIBE_HEADER, avroRecord);
+          }
+        }
+      } catch (Exception e) {
+        LOG.error(String.format("Exception in getting avro field schema types in ScribeParquetAvroConverter: Schema: %s, field: %s, typeName: %s, exception: %s", schema.getName(), fieldName, e));
```

Comment:
You are logging an error. Should you continue?

If you can. Log this message as warn.